libtransmission/announcer.c

```diff
@@ -1434,6 +1508,29 @@ static void on_scrape_done(tr_scrape_response const* response, void* vsession)
         }
     }
 
+    /* Maybe reduce the number of torrents in a multiscrape req */
+    if (multiscrape_too_big(response->errmsg))
+    {
+        char const* url = response->url;
+        int* multiscrape_max = &tr_announcerGetScrapeInfo(announcer, url)->multiscrape_max;
+
+        if (*multiscrape_max == response->row_count)
```

Comment:
We probably want to decrease the limit even if the condition isn't true, using `response->row_count` instead of `*multiscrape_max` in maximum calculation below. Am I missing something?

Reply:
Yes: Transmission can have lots of these in-flight at the same time, so this test is to prevent parallel failures from knocking us down to MIN right off.

This isn't intuitive though. I'll add a comment explaining this.

Reply:
But if we always send less than `TR_MULTISCRAPE_MAX` (which is the default for `tr_scrape_info::multiscrape_max`, per `tr_announcerGetScrapeInfo`, and thus the value of `*multiscrape_max` the first time we get here) to a particular tracker and still get the error, it'll never get reduced, right?

Reply:
That makes sense for the case where the user has fewer than `TR_MULTISCRAPE_MAX` torrents that scrape to a particular tracker.

I think we could handle both cases by testing for `if (*multiscrape_max >= response->row_count)`.

Reply:
Sounds good, yep.

Reply:
Added 957718b68ae7b10b857ececb17cb0b862d8ca2e7